File: 8_kyu/Convert_to_Binary.c
// Converts num into a number printed as decimal looks like the binary format of num.
unsigned long long to_binary(unsigned short num)
{
  long long accu = 0;
  long long bit = 1;
  while( num ) {
    if( num & 1 ) accu += bit;
    bit *= 10;
    num >>= 1;
  }
  return accu;
}

File: 8_kyu/Convert_to_Binary.cs
using System;

public static class Kata
{
  // Converts num into a number printed as decimal looks like the binary format of num.
  public static int ToBinary(int n)
  {
    int accu = 0;
    int bit = 1;
    while( n > 0 ) {
      if( (n & 1) > 0 ) accu += bit;
      bit *= 10;
      n >>= 1;
    }
    return accu;
  }
}
